Source/JavaScriptCore/API/JSAPIGlobalObject.mm

```diff
@@ -152,8 +153,26 @@
     RETURN_IF_EXCEPTION(scope, reject(scope));
 
     auto result = computeValidImportSpecifier(sourceOrigin.url(), specifier);
-    if (result)
-        return import(result.value());
+    if (result) {
+        auto assertions = JSC::retrieveAssertionsFromDynamicImportOptions(globalObject, parameters, { vm.propertyNames->type.impl() });
+        RETURN_IF_EXCEPTION(scope, reject(scope));
+        parameters = jsUndefined();
+        if (!assertions.isEmpty()) {
+            ScriptFetchParameters::Type type = ScriptFetchParameters::Type::None;
+            auto typeString = assertions.get(vm.propertyNames->type.impl());
+            if (typeString == "json"_s)
```

Comment:
It looks a bit inelegant that we have these if else statements in three different places.
Can we share the code somehow?

Reply:
Added `ScriptFetchParameters::parseType` and use it in these places.